parts/linux/cloud-init/nodecustomdata.yml

```diff
@@ -394,7 +394,12 @@ write_files:
     #!/bin/bash
 {{if not IsIPMasqAgentEnabled}}
     {{if IsAzureCNI}}
-    iptables -t nat -A POSTROUTING -m iprange ! --dst-range 168.63.129.16 -m addrtype ! --dst-type local ! -d {{GetParameter "vnetCidr"}} -j MASQUERADE
+    IFS=','
+    read -ra vnetCidrs <<< {{GetParameter "vnetCidr"}}
```

Comment:
for linux we only pass in default vnetcidr

if properties.HostedMasterProfile != nil {
		addValue(parametersMap, "masterSubnet", properties.HostedMasterProfile.Subnet)
		addValue(parametersMap, "vnetCidr", DefaultVNETCIDR)
	}

Reply:
@xuto2, from my understanding, the vnetCidr should be different if the user uses a custom vnet with a different IP address ranges. It looks like that we are lucky because current users are still using all IP address range in 10.0.0.0/8 even if they are using custom vnet. 
I think that Linux also needs to change the code not to use DefaultVNETCIDR for custom vnet.

Reply:
linux use masqAgent so it doesn't matter :)

Reply:
Do you know when IsIPMasqAgentEnabled will be false?

Reply:
ip-masq-agent is enabled in case of Azure CNI by default controlled by toggle, and it's not exposed to our customer. So I think we can regard ipmasq is also enabled for Azure CNI.

Reply:
For SwitftNetworking, it is false.